src/migration/1727870383565-NotificationV2PostAddedUnique.ts

```diff
@@ -0,0 +1,19 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class NotificationV2PostAddedUnique1727870383565
+  implements MigrationInterface
+{
+  name = 'NotificationV2PostAddedUnique1727870383565';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `CREATE UNIQUE INDEX "IDX_notification_v2_post_added_unique" ON "notification_v2" ("referenceId", "referenceType") WHERE type IN ('user_post_added', 'source_post_added', 'squad_post_added')`,
```

Comment:
@rebelchris answering here to your question so its a thread. We only want to dedup post added notification types, if there was no `WHERE` it would also dedup notifications as `post_mention` because they also reference post by id.

Reply:
Yeah makes sense, was just wondering which one allowed a duplicate at all :) 
Don't we actually dedup that one on sending level already? (Think on mention we don't send you the actual post one for instance)

Reply:
We dont but notifications are reused so when you do get one for upvote without where the constraint would block it.

Reply:
Right! yeah that makes sense indeed :)